`rptest`: partially deflake `test_stress_consumer_group_commits`

WillemKauf opened the pull request:
This test can race with adjacent segment compaction while collecting a node's consumer offsets summary, with a backtrace like

```
File "/root/tests/rptest/tests/consumer_group_test.py", line 1690, in test_stress_consumer_group_commits
    summary = olv.consumer_offsets_summary(n)
  File "/root/tests/rptest/clients/offline_log_viewer.py", line 59, in consumer_offsets_summary
    return self._json_cmd(node, "--type consumer_offsets_summary")
  File "/root/tests/rptest/clients/offline_log_viewer.py", line 34, in _json_cmd
    json_out = node.account.ssh_output(cmd, combine_stderr=False)
  File "/opt/.ducktape-venv/lib/python3.10/site-packages/ducktape/cluster/remoteaccount.py", line 41, in wrapper
    return method(self, *args, **kwargs)
  File "/opt/.ducktape-venv/lib/python3.10/site-packages/ducktape/cluster/remoteaccount.py", line 421, in ssh_output
    raise RemoteCommandError(self, cmd, exit_status, stderr.read())
  FileNotFoundError: [Errno 2] No such file or directory: \'/var/lib/redpanda/data/kafka/__consumer_offsets/0_22/2239422-4-v1.log\'\n'
```

Use a `wait_until()` wrapper instead to retry this exceptional case.

The second commit also adds a `LoggingConfig` argument to pass to the `RedpandaService` base class for this test, since it generates large enough logs to fill up the CI agent node's disks.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

* none

WillemKauf committed:
`rptest`: deflake `test_stress_consumer_group_commits`

This test can race with adjacent segment compaction while collecting a
consumer offsets summary, with a backtrace like

```
File "/root/tests/rptest/tests/consumer_group_test.py", line 1690, in test_stress_consumer_group_commits
    summary = olv.consumer_offsets_summary(n)
  File "/root/tests/rptest/clients/offline_log_viewer.py", line 59, in consumer_offsets_summary
    return self._json_cmd(node, "--type consumer_offsets_summary")
  File "/root/tests/rptest/clients/offline_log_viewer.py", line 34, in _json_cmd
    json_out = node.account.ssh_output(cmd, combine_stderr=False)
  File "/opt/.ducktape-venv/lib/python3.10/site-packages/ducktape/cluster/remoteaccount.py", line 41, in wrapper
    return method(self, *args, **kwargs)
  File "/opt/.ducktape-venv/lib/python3.10/site-packages/ducktape/cluster/remoteaccount.py", line 421, in ssh_output
    raise RemoteCommandError(self, cmd, exit_status, stderr.read())
  FileNotFoundError: [Errno 2] No such file or directory: \'/var/lib/redpanda/data/kafka/__consumer_offsets/0_22/2239422-4-v1.log\'\n'
```

Use a `wait_until()` wrapper instead to retry this case.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--6fa2b6f2-a3e6-4d5f-8a5c-9ae06e4116c7-->
<!--TEST RESULTS-->
<!--3397f69c-b79a-47f0-bfcd-a0d93b99d997-->
<!--TEST RESULTS-->
<!--52568e37-2b2b-4728-83ee-44b3e663a2e4-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/66142">build#66142</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------------|:---------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CloudStorageScrubberTest | test_scrubber        | {"cloud_storage_type": 2}                                                                                                                                          | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66142#0196e094-f563-45e9-94b2-d751b93ce66e | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              |
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "enable_failures": false, "mixed_versions": false, "with_chunked_compaction": false, "with_iceberg": false, "with_tiered_storage": true} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66142#0196e094-f561-49b6-bc90-444930c27f5c | FLAKY         | 20/21    | upstream reliability is '97.09302325581395'. current run reliability is '95.23809523809523'. drift is 1.85493 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/66150">build#66150</a></summary>


| test_class                 | test_method                                       | test_arguments                                                                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:---------------------------|:--------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| PartitionReassignmentsTest | test_add_partitions_with_inprogress_reassignments |                                                                                                                                                                    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66150#0196e35b-9b85-4941-82f4-01faeed2a2e3 | FLAKY         | 17/21    | upstream reliability is '90.14084507042254'. current run reliability is '80.95238095238095'. drift is 9.18846 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest   | test_node_operations                              | {"cloud_storage_type": 2, "enable_failures": false, "mixed_versions": false, "with_chunked_compaction": false, "with_iceberg": false, "with_tiered_storage": true} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66150#0196e35b-9b84-49bd-9115-42aaf93a9ca5 | FLAKY         | 20/21    | upstream reliability is '97.09302325581395'. current run reliability is '95.23809523809523'. drift is 1.85493 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/66155">build#66155</a></summary>


| test_class                | test_method                              | test_arguments                                                                                                                                                      | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:--------------------------|:-----------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest     | test_migrated_topic_data_integrity       | {"params": {"cancellation": null, "use_alias": false}, "transfer_leadership": true}                                                                                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66155#0196e592-0006-411a-b367-26d86e31cc0d | FLAKY         | 20/21    | upstream reliability is '98.19494584837545'. current run reliability is '95.23809523809523'. drift is 2.95685 and the allowed drift is set to 50. The test should PASS  |
| RestCatalogConnectionTest | test_redpanda_connection_to_rest_catalog | {"cloud_storage_type": 1}                                                                                                                                           | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66155#0196e592-0006-411a-b367-26d86e31cc0d | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               |
| DeleteRecordsKafkaTest    | test_delete_records_non_empty_topic      | {"truncate_point": "start_offset"}                                                                                                                                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66155#0196e592-0006-411a-b367-26d86e31cc0d | FLAKY         | 20/21    | upstream reliability is '98.72881355932203'. current run reliability is '95.23809523809523'. drift is 3.49072 and the allowed drift is set to 50. The test should PASS  |
| RandomNodeOperationsTest  | test_node_operations                     | {"cloud_storage_type": 2, "enable_failures": false, "mixed_versions": false, "with_chunked_compaction": false, "with_iceberg": false, "with_tiered_storage": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66155#0196e533-cf3c-4e90-9a5d-21904220f8e2 | FLAKY         | 20/21    | upstream reliability is '97.6878612716763'. current run reliability is '95.23809523809523'. drift is 2.44977 and the allowed drift is set to 50. The test should PASS   |
| DisablingPartitionsTest   | test_disable                             |                                                                                                                                                                     | ducktape    | https://buildkite.com/redpanda/redpanda/builds/66155#0196e592-0005-4aad-8909-a12e3a6b40f4 | FLAKY         | 14/21    | upstream reliability is '85.84070796460178'. current run reliability is '66.66666666666666'. drift is 19.17404 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/consumer_group_test.py::ConsumerGroupOffsetResetTest.test_stress_consumer_group_commits

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/consumer_group_test.py::ConsumerGroupOffsetResetTest.test_stress_consumer_group_commits

WillemKauf committed:
`rptest`: Use `LoggingConfig` in `consumer_group_test.py`

This test could generate enough output to fill up the agent nodes in CI:

```
[WARNING - 2025-05-18 15:24:36,604 - test - copy_service_logs - lineno:162]: Error copying log backtraces from
{'path': '/var/lib/redpanda/redpanda_backtrace.log', 'collect_default': True} to /build/tests/results/2025-05-18--002/ConsumerGroupOffsetResetTest/test_stress_consumer_group_commits/15/RedpandaService-0-140178931023632/docker-rp-18.service <RedpandaService-0-140178931023632: num_nodes: 3, nodes: ['docker-rp-16', 'docker-rp-17', 'docker-rp-18']>: [Errno 28] No space left on device
```

Use a custom `LoggingConfig` to reduce the amount of output generated.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/66154-->
<!--9fef8f9c-1de0-4588-ba4e-04947cb8c7de-->
<!--title-->
## Retry command for [Build#66154](https://buildkite.com/redpanda/redpanda/builds/66154)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/consumer_group_test.py::ConsumerGroupOffsetResetTest.test_stress_consumer_group_commits
```
<!--/body-->

bashtanov commented on tests/rptest/tests/consumer_group_test.py:
```diff
@@ -1700,3 +1710,14 @@ def test_stress_consumer_group_commits(self):
+                lambda: get_summary(n),
+                timeout_sec=60,
+                backoff_sec=1,
+                retry_on_exc=True,
```
This way we catch the asserts above. Are they unlikely to be transient?

WillemKauf replied:
Yikes good point. I forget that `assert`s in Python just throw. i'll have a follow-up fix to this.

WillemKauf replied:
Though I doubt these errors are transient, we should fail the test on the first sighting of a failed assert.

WillemKauf replied:
https://github.com/redpanda-data/redpanda/pull/26213
